recaptcha_enterprise/demosite/src/main/java/app/MainController.java

```diff
@@ -0,0 +1,102 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package app;
+
+import java.util.LinkedHashMap;
+import org.json.JSONObject;
+import org.json.JSONTokener;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.servlet.ModelAndView;
+import recaptcha.CreateAssessment;
+
+@RestController
+@RequestMapping
+public class MainController {
+
+  private static final LinkedHashMap<String, String> CONTEXT = new LinkedHashMap<>();
+
+  static {
+    CONTEXT.put("project_id", System.getenv("GOOGLE_CLOUD_PROJECT"));
+    CONTEXT.put("site_key", System.getenv("SITE_KEY"));
+  }
+
+  @GetMapping(value = "/")
+  public static ModelAndView home() {
+    return new ModelAndView("home", CONTEXT);
+  }
+
+  @GetMapping(value = "/store")
+  public static ModelAndView store() {
+    return new ModelAndView("store", CONTEXT);
+  }
+
+  @GetMapping(value = "/login")
+  public static ModelAndView login() {
+    return new ModelAndView("login", CONTEXT);
+  }
+
+  @GetMapping(value = "/comment")
+  public static ModelAndView comment() {
+    return new ModelAndView("comment", CONTEXT);
+  }
+
+  @GetMapping(value = "/signup")
+  public static ModelAndView signup() {
+    return new ModelAndView("signup", CONTEXT);
+  }
+
+  @GetMapping(value = "/game")
+  public static ModelAndView game() {
+    return new ModelAndView("game", CONTEXT);
+  }
+
+  @PostMapping(value = "/create_assessment", produces = "application/json")
+  public static @ResponseBody
+  ResponseEntity<String> createAssessment(@RequestBody String json) {
+    String projectId = System.getenv("GOOGLE_CLOUD_PROJECT");
+    JSONObject jsonObject = new JSONObject(new JSONTokener(json));
+    JSONObject credentials = jsonObject.getJSONObject("recaptcha_cred");
+    JSONObject result;
+
+    try {
+      // <!-- ATTENTION: reCAPTCHA Example (Server Part 1/2) Starts -->
+      result = CreateAssessment.createAssessment(projectId,
+          CONTEXT.get("site_key"),
+          credentials.getString("token"),
+          credentials.getString("action"));
+      // <!-- ATTENTION: reCAPTCHA Example (Server Part 1/2) Ends -->
+    } catch (Exception e) {
+      result = new JSONObject()
```

Comment:
Demosite provides customers the functionality to edit the code and do a live_reload (as defined in the dockerfile). When executing this functionality, there might be errors thrown from the backend which we might need to display to the customers (in place of the result). 
Hence, whatever be the response from the backend, it's sent as an OK response. wdyt?

Reply:
It's really hard to connect all those pieces. From what I can tell this will silently fail and be ignored in `showAssessmentInDemo`. We should be careful about blurring the lines of what an actual user should do and what this demo is showing.

Reply:
I understand. Thanks for the suggestion.
I agree with it and I've modified the call to include appropriate Status information. We can do the heavy-lifting on the client side to decide if the error message should be shown to the user.